io/net/port_bounce.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+# Copyright: 2017 IBM
+# Author: Pridhiviraj Paidipeddi <ppaidipe@linux.vnet.ibm.com>
+# this script runs portbounce test on different ports of fc or fcoe switches.
+
+import time
+import telnetlib
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+
+
+class CommandFailed(Exception):
+    def __init__(self, command, output, exitcode):
+        self.command = command
+        self.output = output
+        self.exitcode = exitcode
+
+    def __str__(self):
+        return "Command '%s' exited with %d.\nOutput:\n%s" \
+               % (self.command, self.exitcode, self.output)
+
+
+SHORT_BOUNCE_TIME = 5
+LONG_BOUNCE_TIME = 250
+
+
+class PortBounceTest(Test):
+
+    """
+    :param type: <fc/fcoe> : type of switch fc/fcoe <in small case>
+    :param fcoe_fc: <yes/no> : If port is an FC port in FCOE switch
+    :param switch_name: FC Switch name/ip
+    :param userid: FC switch user name to login
+    :param password: FC switch password to login
+    :param port_ids: FC switch port ids where port needs to disable/enable
+    :param count: Number of times test to run
+    """
+
+    def setUp(self):
+        """
+        test parameters
+        """
+        self.parameters()
+
+    def parameters(self):
+        self.fc_type = self.params.get("type", '*', default=None)
+        self.fcoe_fc = self.params.get("fcoe_fc", '*', default=None)
+        self.switch_name = self.params.get("switch_name", '*', default=None)
+        self.userid = self.params.get("userid", '*', default=None)
+        self.password = self.params.get("password", '*', default=None)
+        port_ids = self.params.get("port_ids", '*', default=None)
```

Comment:
can this be like 
`self.port_ids =  self.params.get("port_ids", '*', default=None).split(",")` want to eliminate extra variable as it make GC more work to do